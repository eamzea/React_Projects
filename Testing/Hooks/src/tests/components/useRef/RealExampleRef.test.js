import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import RealExampleRef from "../../../components/useRef/RealExampleRef";

describe("Pruebas en <RealExampleRed/>", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("Debe mostrar el componente <MultipleCustomHooks/>", () => {
    wrapper.find("Button").simulate("click");

    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
