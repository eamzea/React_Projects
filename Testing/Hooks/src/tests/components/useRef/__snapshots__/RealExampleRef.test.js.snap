// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRed/> Debe mostrarse correctamente 1`] = `
<div>
  <p
    className="h1"
  >
    Real Example Ref
  </p>
  <hr />
  <Button
    active={false}
    className="mt-5"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Show/Hide
  </Button>
</div>
`;
